cmd/dogstatsd/main.go

```diff
@@ -225,8 +225,15 @@ func stopAgent(ctx context.Context, cancel context.CancelFunc) {
 	// gracefully shut down any component
 	cancel()
 
-	metaScheduler.Stop()
-	statsd.Stop()
+	// stop metaScheduler and statsd if they are initiated
```

Comment:
```suggestion
	// stop metaScheduler and statsd if they are instantiated
```
Or you can just remove the comment as the previous one is already explicit ("gracefully...").